### ISSUE TITLE
Rename default "master" zone name into "main"

### ISSUE THREAD

**Issue by Mikesch-mp** [Icinga/icinga2#8689]
## Is your feature request related to a problem? Please describe.

Since master is counted as a racism term now, it should be replaced.

## Describe the solution you'd like

Choose a term like "main"

## Describe alternatives you've considered

There is no alternative, just dont use racism terms anymore

## Additional context

Github for example did the change and many other projects that uses those old (racism) terms did it too. Just google it

**Comment by Al2Klimov** [comment]
Sure, e.g. according to https://en.wiktionary.org/wiki/master **one possible meaning** is "The owner of an animal or slave". But it’s not the only one and even that meaning does not exclude the possibility an animal owner is meant actually. So tbh I consider this issue as "works for me".

But today I'm kind and will ask for a second opinion...

**Comment by julianbrost** [comment]
+1 for reconsidering the naming. When doing so, we should aim for more distinctive names as the term master is currently used for many different things:
* git master branch
* master zone (zone without a parent)
* master node (node in a master zone)
* config master (one node in the master zone which has the authority over the config files)
* zone master (one node in each zone determined internally by Icinga, usually not visible to users)

**Comment by lbetz** [comment]
I am currently working on a new edition of the Icinga book, which is also being translated into English, and I decided on the following names:

Zone (zone without parent) main
"Icinga" Server (endpoint of zone main)
"Icinga" Config Server for the Server that hosts the configuration files

I also dropped the word satellite from all texts and replaced it with worker. But that's only because I take the word written as unesthetic. 

For puppet-icinga (https://github.com/Icinga/puppet-icinga) and the icinga-installer project (https://github.com/netways/icinga-installer) I also used this naming.

**Comment by Al2Klimov** [comment]
Excuse me... you... _decided_? This remembers me of _someone_ who _decided_ on a name for a particular zone their tool had required. Later more and more users complained _ugh, that zone isn’t present_ and we had to ship that zone in Icinga 2 with the name already been _decided_.

Please do us core devs and every of our successors a favor and _discuss_ those names instead (btw. I opt for s/main/**root**/ and s/Icinga Server/**Icinga Root Server**/) once we’ve discussed whether we "fix" this "issue" (btw. still waiting for feedback from the assignee). Yes, **whether**, not when. In opposite to some of us, for me this isn’t "just doing things for the sake of doing things". For me this is a complete waste of time.

Sure, **slave**ry is a bad thing (btw. not only for blacks (USA), but also e.g. for Russians (Chechnya) and Soviets (Gulag)) and we shall not use the word "slave" in code+docs, but... _master_? _Satellite_? Seriously?

I'm open to change my mind if you give me some actual arguments and reasonings. For the record: "Just google it" **is not** an argument.

**Comment by lippserd** [comment]
I second that renaming master is required. But we use the word satellite everywhere in the Icinga ecosystem and I don‘t see why we should change that as well.

**Comment by lbetz** [comment]
Yes, I decided to write a book, I decided the content and I decided not to call a configurable zone master. The text for the book was written before you started talking about this topic, so it is older than this ticket.

Of course, there is no need to change the satellite, I should have also mentioned that I use worker aka satellite in the text.

**Comment by lippserd** [comment]
Between `main` and `root` I choose `main` because I cannot guarantee that current `masters` are always the root nodes.
Some time ago we had the idea of setting up the web nodes as a zone above the master zone for functions such as automatic node detection and uniform API and CLI.
I would like a word that does not require the "node" suffix. `Icinga Master` works independently, while `Icinga Main` and `Icinga Root` always need `Node`.

**Comment by lbetz** [comment]
I also didn't want to start a discussion about satellite as a word, it was just unnecessary information as I now see.

**Comment by lbetz** [comment]
I also prefer main, but root actually describes it better, because there is no further parent zone. I am running some installations with a parent zone for Icinga Web and I named it master in the past.
I would have liked root better here, but it is a matter of definition, if you define main / root as the zone that does not have a parent, any further name for an additional parent would be wrong.

**Comment by Al2Klimov** [comment]
> Between `main` and `root` I choose `main` because I cannot guarantee that current `masters` are always the root nodes.
> Some time ago we had the idea of setting up the web nodes as a zone above the master zone for functions such as automatic node detection and uniform API and CLI.

Nice technical concept, but with one little problem: it depends on the future – which may or may not stay intact. As long as this future idea stays just an idea, I'd not worry too much about this. After all the top-level DNS servers are called _root servers_ as well, so this naming should be familiar in a tree-ish hierarchy.
